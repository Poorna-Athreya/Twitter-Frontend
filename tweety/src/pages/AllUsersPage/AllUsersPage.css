.all-users-page{
    margin:20px;
    
    height: 80vh;
} 