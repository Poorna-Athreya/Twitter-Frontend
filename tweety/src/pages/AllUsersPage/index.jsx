import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { AllUsers } from '../../components';
import './AllUsersPage.css';
import makeRequest from '../../utils/makeRequest';
import { getUsers } from '../../constants/apiEndpoints';

function AllUsersPage() {
  const [allUsers, setAllUsers] = useState([]);
  const navigate = useNavigate();
  useEffect(() => {
    makeRequest(getUsers, {}, navigate).then((responseData) => {
      setAllUsers(responseData.users);
    });
  }, []);
  return (
    <div className="all-users-page">
      <AllUsers allUsers={allUsers} />
    </div>
  );
}
export default AllUsersPage;
