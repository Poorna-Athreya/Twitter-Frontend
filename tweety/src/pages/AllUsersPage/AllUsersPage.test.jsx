import React from 'react';
import {
  render, screen,
} from '@testing-library/react';
import AllUsersPage from './index';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({ useNavigate: () => mockNavigate }));

const mockMakeRequest = () => ({
  users: [
    {
      id: 110,
      name: 'Twitter Dev',
      handle: 'TwitterDev',
      location: 'Internet',
    },
    {
      id: 111,
      name: 'Lorem Ipsum',
      handle: 'LoremIpsum',
      location: 'Internet',
    }],
});
jest.mock('../../utils/makeRequest/', () => () => {
  const response = mockMakeRequest();
  return Promise.resolve(response);
});

describe('AllUsersPage', () => {
  it('should take a snapshot of AllUsersPage', () => {
    const { asFragment } = render(<AllUsersPage />);
    expect(asFragment(<AllUsersPage />)).toMatchSnapshot();
  });
  it('should render all the user details when the page is loaded', async () => {
    render(<AllUsersPage />);
    expect(await screen.findAllByText('Twitter Dev')).toBeTruthy();
    expect(await screen.findAllByText('@TwitterDev')).toBeTruthy();
    expect(await screen.findAllByText('Internet')).toBeTruthy();
    expect(await screen.findAllByText('Lorem Ipsum')).toBeTruthy();
    expect(await screen.findAllByText('@LoremIpsum')).toBeTruthy();
  });
});
