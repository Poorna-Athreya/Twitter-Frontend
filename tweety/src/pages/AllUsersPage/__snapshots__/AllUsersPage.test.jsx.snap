// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllUsersPage should take a snapshot of AllUsersPage 1`] = `
<DocumentFragment>
  <div
    class="all-users-page"
  >
    <div
      class="users-container"
    />
  </div>
</DocumentFragment>
`;
